File: medplum-typescript/packages/sdk/src/model/fhir/Invoice.ts
import { Annotation } from "./Annotation";
import { code } from "./code";
import { CodeableConcept } from "./CodeableConcept";
import { dateTime } from "./dateTime";
import { Extension } from "./Extension";
import { id } from "./id";
import { Identifier } from "./Identifier";
import { InvoiceStatus } from "./InvoiceStatus";
import { Invoice_LineItem } from "./Invoice_LineItem";
import { Invoice_Participant } from "./Invoice_Participant";
import { Invoice_PriceComponent } from "./Invoice_PriceComponent";
import { markdown } from "./markdown";
import { Meta } from "./Meta";
import { Money } from "./Money";
import { Narrative } from "./Narrative";
import { Reference } from "./Reference";
import { ResourceList } from "./ResourceList";
import { uri } from "./uri";

export interface Invoice {
    resourceType: string;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    text: Narrative | null | undefined;
    contained: ResourceList[] | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    identifier: Identifier[] | null | undefined;
    status: InvoiceStatus | null | undefined;
    cancelledReason: string | null | undefined;
    type: CodeableConcept | null | undefined;
    subject: Reference | null | undefined;
    recipient: Reference | null | undefined;
    date: dateTime | null | undefined;
    participant: Invoice_Participant[] | null | undefined;
    issuer: Reference | null | undefined;
    account: Reference | null | undefined;
    lineItem: Invoice_LineItem[] | null | undefined;
    totalPriceComponent: Invoice_PriceComponent[] | null | undefined;
    totalNet: Money | null | undefined;
    totalGross: Money | null | undefined;
    paymentTerms: markdown | null | undefined;
    note: Annotation[] | null | undefined;
}

File: medplum-java/medplum-model/src/main/java/com/fhir/Invoice.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableInvoice.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface Invoice {
  Optional<List<Annotation>> note();

  Optional<Reference> account();

  String resourceType();

  Optional<Money> totalNet();

  Optional<String> cancelledReason();

  Optional<Reference> issuer();

  Optional<Reference> subject();

  Optional<List<Invoice_LineItem>> lineItem();

  Optional<List<Identifier>> identifier();

  Optional<List<Extension>> modifierExtension();

  Optional<code> language();

  Optional<CodeableConcept> type();

  Optional<List<Extension>> extension();

  Optional<Narrative> text();

  Optional<List<ResourceList>> contained();

  Optional<Reference> recipient();

  Optional<Money> totalGross();

  Optional<markdown> paymentTerms();

  Optional<uri> implicitRules();

  Optional<id> id();

  Optional<Meta> meta();

  Optional<InvoiceStatus> status();

  Optional<List<Invoice_PriceComponent>> totalPriceComponent();

  Optional<dateTime> date();

  Optional<List<Invoice_Participant>> participant();

  static ImmutableInvoice.ResourceTypeBuildStage builder() {
    return ImmutableInvoice.builder();
  }
}
